### ISSUE TITLE
[URGENT] [New version] Bots completely stop moving randomly

### ISSUE THREAD

**Issue by Samg381** [EfeDursun125/TF2-EBOTS#11]
After respawning several times, bots will stop moving in spawn. I have built navmesh.

**Comment by Samg381** [comment]
I think the bots stuck walking in circles on capture point might be caused by this line:

https://github.com/EfeDursun125/TF2-EBOTS/blob/d0b609030b60c5ce30044203abf5628c5ed8200d/NewVersion/include/ebotai/goap/defaultai.inc#L772C84-L772C84

**Comment by EfeDursun125** [comment]
bots might cannot find path due of broken autogenerated navmesh, new version is not really desinged to be used with navmesh, but you can use navmesh + waypoints at the same time it might help bots navigate. like sniper, engineer.

example of this is used on koth_suijin map, you can take a look.

use the command ebot_waypoint_on to see waypoints, use ebot_waypoint_add command to add waypoints, after that set their flags 2 for sniper, 3 for defend, 12 for sentry, 14 for teleporter enter, 15 for teleporter exit and 16 for capture point (only use this in koth_ and cp_ maps)

use ebot_waypoint_set_team to give specific team only, 2 = red and 3 = blu

these are used in cp & payload maps which they have multiple control points, it determines does waypoint active when current control point is capturable
ebot_waypoint_set_area
ebot_waypoint_add_area
ebot_waypoint_remove_area

payload maps starts with first capture point right? then first area is 1, and last one is 5 (total control point count in the map) same applies for cp maps too

if you want change waypoint's aim direction use these
ebot_waypoint_set_aim_start
ebot_waypoint_set_aim_end

determines how closely the bot can use the waypoint, default is 8 units
ebot_waypoint_add_radius

when you finish adding navpoints for make bots know what they can do use this command
ebot_waypoint_save_nav

REMEMBER DO NOT USE ebot_waypoint_save, USE ebot_waypoint_save_nav

and saved navpoint file will appear at tf\addons\sourcemod\ebot\waypoints\nav
normal waypoints are saved to tf\addons\sourcemod\ebot\waypoints

**Comment by EfeDursun125** [comment]
dont forget, this is not fully waypointing process, only add specific ones like sentry, sniper and capture point waypoints, navmesh still will be used.

**Comment by Samg381** [comment]
Thanks. If a bot respawns in the spawn room and doesn't move, what command should I run?

**Comment by EfeDursun125** [comment]
which class? it was all?

**Comment by Samg381** [comment]
It seems to be random. I have seen it happen with every class. Sometimes the map will start and a few of them will not start moving. Other times, bots will start working normally, and then stop moving when they respawn. Really strange behavior.

**Comment by EfeDursun125** [comment]
which gamemode, which map? if its caused by red team while they're defending you can use defend waypoints

**Comment by Samg381** [comment]
CTF on 2fort, doublecross, turbine (every map I've tried)

**Comment by Samg381** [comment]
Some feedback regarding the new waypoint system: I removed the ebot/waypoints folder from the sourcemod directory. The plugin appears to now be using the old waypoint system, which actually seems to be working a lot better in most cases. When using the new waypoint system, bots would get stuck in several areas (for example at the top of the stairs near spawn in the courtyard in 2fort)

Do you have instructions for editing waypoints using the old navigation system? I can still see the waypoints and the navigation path illuminated on the ground, although it is less detailed than the newer plugin. Are there commands for editing the old waypoints?

**Comment by EfeDursun125** [comment]
there's no old navigation system.
there's three types of navigation system: 
1. Fully Waypoints (The Best, Maximixes the bot's performance (not the fps), navmesh isn't used.)
2. Half Waypoints/Navpoints (waypoints determines bot goals, like sniper/sentry/defend, navmesh is used for find path to these waypoints)
3. Navmesh (The Worst, bots act as a fool and they perform bad, because without waypoints they don't know best spots/points/areas)


ebot_waypoint_on shows the waypoints

ebot_waypoint_save saves full waypoints
ebot_waypoint_save_nav saves half waypoints/navpoints

old version of ebot performs much more better than new ebots on navmesh because they desinged to be used with navmesh.
you made a bad choice by removing waypoints instead you can fix waypoints wtih editing.

**Comment by Samg381** [comment]
Thanks for the info. To be clear, I have always been using the **new** version of ebots. If my understanding is correct, I have only tried the `navmesh` and `fully wavepoints` modes you listed above since I had removed/installed the waypoints folder.

Regarding your last statement: when testing the `fully waypoints` mode, I _did_ try for several hours to edit the waypoints using the commands you provided before I uninstalled the folder to revert back to `navmesh` mode. Despite my efforts, the bots would get stuck in certain areas that did not correspond to any nearby waypoints (shown with ebot_waypoint_on). One example of this would be bots getting stuck at the top of the stairs near spawn in the courtyard in 2fort. There was no waypoint / area to edit / remove that would fix this. They would walk to that point and stay there even if I had removed all nearby waypoints. This is why I removed the waypoints folder to revert to `navmesh` mode.

Another issue with the `fully waypoints` mode is the jerky aim behavior. It seems as if the look direction can be customized with the commands, however the movement is not as fluid as with `navmesh` and has caused users to notice the bots more rapidly.
